### PR TITLE
Fix the SONiC version starting with HEAD issue

### PR DESCRIPTION
**Pull request by xumia**
#### Why I did it
Fix the SONiC version starting with HEAD issue. For example, sonic-broadcom-HEAD.40620469-1d98d18cf.bin.

#### How I did it
Set the correct branch name after the code checked out.

#### How to verify it
After applied the change "git checkout -b 201911", the version is good.
```
$ git branch
* (HEAD detached at cf54cbc5)
$ sonic_get_version
HEAD.0-cf54cbc5
$ git checkout -b 201911
Switched to a new branch '201911'
$ sonic_get_version     
201911.0-cf54cbc5
```
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

**Comment by qiluo-msft** [comment]
The change looks good. Seem the PR description "How to verify it" section is not relevant to this PR. Could you check?

**Review by qiluo-msft — COMMENTED** [review]
LGTM

**Comment by xumia** [comment]
> The change looks good. Seem the PR description "How to verify it" section is not relevant to this PR. Could you check?

@qiluo-msft , fixed the description, thanks.

**Review by qiluo-msft — APPROVED** [review]

File: .azure-pipelines/azure-pipelines-image-template.yml [hunk 1]
```diff
@@ -38,6 +38,7 @@ jobs:
         submodules: recursive
         displayName: 'Checkout code'
       - script: |
+          git checkout -b $(Build.SourceBranchName)
```
Comment (lguohan) [inline .azure-pipelines/azure-pipelines-image-template.yml:41]: i have some concern on this one. the checkout behavior is complicated when we deal will pr build. we should only fetch the branch, but should not modify the checkout behavior. this won't work for pr build.

Comment (xumia) [reply .azure-pipelines/azure-pipelines-image-template.yml:41]: There is a -b option, means to create a new branch. It should not change the current commit.
git checkout -b <branchname>
Different from: 
git checkout <branchname>

Comment (xumia) [reply .azure-pipelines/azure-pipelines-image-template.yml:41]: The branch name is merge, and the commit not change see detail in draft PR: https://github.com/Azure/sonic-buildimage/pull/7329 and the build result: https://dev.azure.com/mssonic/build/_build/results?buildId=10941&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=1f6d5e21-f0ce-508d-13df-d1c50f7ea7b9

* (HEAD detached at pull/7329/merge) 482f951a Merge c2486425cac936b5ce1834a433922226fc570377 into 15487f4b5e48d9d5dcf3433997be30beb8d33c03
SourceBranchName=merge
Switched to a new branch 'merge'
* merge 482f951a Merge c2486425cac936b5ce1834a433922226fc570377 into 15487f4b5e48d9d5dcf3433997be30beb8d33c03

Comment (lguohan) [reply .azure-pipelines/azure-pipelines-image-template.yml:41]: the branch name "merge" is not correct one.

**Pull request merged**